src/Standards/PSR2/Sniffs/Namespaces/UseDeclarationSniff.php

```diff
@@ -138,48 +138,48 @@ public function process(File $phpcsFile, $stackPtr)
         }
 
         $end = $phpcsFile->findNext(T_SEMICOLON, ($stackPtr + 1));
-        $candidateComment = $end;
-
-        do {
-            $nextComment      = $candidateComment;
-            $candidateComment = $phpcsFile->findNext(Tokens::$emptyTokens, ($nextComment + 1));
-        } while ($candidateComment !== false && $tokens[$candidateComment]['line'] === $tokens[$end]['line']);
-
-        $end = $nextComment;
-
         if ($end === false) {
             return;
         }
 
-        $next = $phpcsFile->findNext(T_WHITESPACE, ($end + 1), null, true);
+        // Find either the start of the next line or the beginning of the next statement,
+        // whichever comes first.
+        for ($end = ++$end; $end < $phpcsFile->numTokens; $end++) {
+            if (isset(Tokens::$emptyTokens[$tokens[$end]['code']]) === false) {
```

Comment:
It seems like some of the tokens in `Tokens::$emptyTokens` shouldn't be allowed: I think that only inline comments should be allowed and multi-line comments shouldn't be: https://github.com/dhensby/PHP_CodeSniffer/blob/master/src/Util/Tokens.php#L435

I don't think stuff like this should pass...
```
use Foo\Bar; /*
 * This multi-line comment shouldn't be allowed here.
 */
```

So, this sniff should build and use an array of `$inlineCommentTokens`.

@jrfnl @dhensby Do you agree?

Reply:
> I don't think stuff like this should pass...

Have you tested it ? Not sure about previously, but this will definitely throw an error with the code I submitted.

Though I have to admit, I think the fixer will fail on it. I'll add an additional unit test for that and have a go at getting the fixer sorted.

> So, this sniff should build and use an array of $inlineCommentTokens.
> @jrfnl @dhensby Do you agree?

No. Multi-line block comments are tokenized as `T_COMMENT` as well, so that wouldn't solve anything.

Reply:
@jrfnl I'll admit I didn't test it. Sorry! I hadn't realized there wasn't a separate token for inline comments.